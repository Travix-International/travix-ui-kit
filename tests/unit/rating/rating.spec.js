import { mount } from 'enzyme';
import React from 'react';
import Rating from '../../../components/rating/rating';

describe('Rating', () => {
  describe('#render()', () => {
    it('renders with default props and mods provided', () => {
      const renderTree = mount(
        <Rating rate={0} />
      );

      const ratingElement = renderTree.find('.ui-rating');
      const valueElement = ratingElement.find('.ui-rating-value');
      const ratingStars = ratingElement.children('b');
      const valueStars = valueElement.children('b');

      expect(renderTree).toMatchSnapshot();
      expect(ratingStars.length).toEqual(5);
      expect(valueStars.length).toEqual(5);
      expect(valueElement.instance().style.width).toEqual("0%");
    });

    it('renders with custom data attributes', () => {
      const dataAttrs = {
        'ui-test': 'rating',
      };

      const renderTree = mount(
        <Rating dataAttrs={dataAttrs} rate={0} />
      );

      expect(renderTree).toMatchSnapshot();
    });

    it('renders with custom props', () => {
      const size = 10;
      const renderTree = mount(
        <Rating rate={1.5} size={size} />
      );

      const ratingElement = renderTree.find('.ui-rating');
      const valueElement = ratingElement.find('.ui-rating-value');
      const ratingStars = ratingElement.children('b');
      const valueStars = valueElement.children('b');

      expect(renderTree).toMatchSnapshot();
      expect(ratingStars.length).toEqual(size);
      expect(valueStars.length).toEqual(size);
      expect(valueElement.instance().style.width).toEqual("15%");
    });
  });
});
