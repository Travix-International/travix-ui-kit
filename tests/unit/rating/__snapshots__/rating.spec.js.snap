// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating #render() renders with custom data attributes 1`] = `
<Rating
  dataAttrs={
    Object {
      "ui-test": "rating",
    }
  }
  rate={0}
  size={5}
>
  <div
    className="ui-rating"
    data-ui-test="rating"
  >
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <div
      className="ui-rating-value"
      style={
        Object {
          "width": "0%",
        }
      }
    >
      <b>
        ★
      </b>
      <b>
        ★
      </b>
      <b>
        ★
      </b>
      <b>
        ★
      </b>
      <b>
        ★
      </b>
    </div>
  </div>
</Rating>
`;

exports[`Rating #render() renders with custom props 1`] = `
<Rating
  rate={1.5}
  size={10}
>
  <div
    className="ui-rating"
  >
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <div
      className="ui-rating-value"
      style={
        Object {
          "width": "15%",
        }
      }
    >
      <b>
        ★
      </b>
      <b>
        ★
      </b>
      <b>
        ★
      </b>
      <b>
        ★
      </b>
      <b>
        ★
      </b>
      <b>
        ★
      </b>
      <b>
        ★
      </b>
      <b>
        ★
      </b>
      <b>
        ★
      </b>
      <b>
        ★
      </b>
    </div>
  </div>
</Rating>
`;

exports[`Rating #render() renders with default props and mods provided 1`] = `
<Rating
  rate={0}
  size={5}
>
  <div
    className="ui-rating"
  >
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <b>
      ★
    </b>
    <div
      className="ui-rating-value"
      style={
        Object {
          "width": "0%",
        }
      }
    >
      <b>
        ★
      </b>
      <b>
        ★
      </b>
      <b>
        ★
      </b>
      <b>
        ★
      </b>
      <b>
        ★
      </b>
    </div>
  </div>
</Rating>
`;
